ttg/ttg/edge.h

```diff
@@ -103,6 +130,10 @@ namespace ttg {
       return result;
     }
 
+    bool is_pull_edge() const {
+      return p.at(0)->is_pull_edge;
```

Comment:
Should/can we handle the case where an Edge represents multiple edges? (https://github.com/TESSEorg/ttg/pull/191/files#diff-428ea16d89d546c44e669e55118f6af786a3339aa972f5aafcd17bc03d40bb48R116)

Reply:
We should probably limit pull terminals to be composed using single edges only.

Reply:
I remember we talked about that a while back. It would be nice to be able to merge pull and regular terminals but it was not immediately clear how to make sure when to pull from a terminal and when to wait for a value. A predicate might solve that issue (akin to making `is_pull_edge` dependent on a `key`). We should at least throw an exception in the multi-Edge constructor if one of the edges is a pull terminal.

Reply:
Agree. I added a check in the multi-edge constructor.